app/models/form_response.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+class InvalidStatusError < StandardError; end
+
+class FormResponse < ApplicationRecord
+  include Discard::Model
+
+  default_scope -> { kept }
+
+  belongs_to :form
+  belongs_to :created_by, class_name: 'User'
+  belongs_to :submitted_by, class_name: 'User', optional: true
+
+  has_one :step_orderable_element, as: :orderable, dependent: :destroy
+
+  enum :status, { pending: 0, submitted: 1, locked: 2 }
+
+  has_many :form_field_values, dependent: :destroy
+
+  def step
+    step_orderable_element&.step
+  end
+
+  def parent
+    if step_orderable_element
+      protocol = step_orderable_element.step.protocol
+      protocol.my_module || protocol
```

Comment:
I think parent must be step. Why we check it is my_module or protoco?
In controller when we used word parent it reference to Step